library/core/src/fmt/builders.rs

```diff
@@ -134,7 +137,13 @@ impl<'a, 'b: 'a> DebugStruct<'a, 'b> {
                 value.fmt(&mut writer)?;
                 writer.write_str(",\n")
             } else {
-                let prefix = if self.has_fields { ", " } else { " { " };
+                let prefix = if self.has_fields {
+                    ", "
+                } else if !self.has_name {
+                    "{ "
+                } else {
+                    " { "
+                };
```

Comment:
I don't think it's worth this amount of code complication. It's a helper for making your formatter print structs the way the stdlib's impls print structs.